.travis.yml

```diff
@@ -0,0 +1,28 @@
+git:
+  depth: false  # fetch all branches
+language: python
+services: [docker]
+matrix:
+  include:
+    - env: TARGET=astyle/Dockerfile.tested
+    - env: TARGET=autopep8/Dockerfile.tested
+    - env: TARGET=black/Dockerfile.tested
+    - env: TARGET=brittany/Dockerfile.tested
+    - env: TARGET=dfmt/Dockerfile.tested
+    - env: TARGET=elm-format/Dockerfile.tested
+    - env: TARGET=google-java-format/Dockerfile.tested
+    - env: TARGET=hindent/Dockerfile.tested
+    - env: TARGET=hlint/Dockerfile.tested
+    - env: TARGET=pg_format/Dockerfile.tested
+    - env: TARGET=php-cs-fixer/Dockerfile.tested
+    - env: TARGET=prettier/Dockerfile.tested
+    - env: TARGET=reorder-python-imports/Dockerfile.tested
+    - env: TARGET=rubocop/Dockerfile.tested
+    - env: TARGET=rustfmt/Dockerfile.tested
+    - env: TARGET=shfmt/Dockerfile.tested
+    - env: TARGET=stylish-haskell/Dockerfile.tested
+    - env: TARGET=terraform/Dockerfile.tested
+    - env: TARGET=yapf/Dockerfile.tested
+    - env: TARGET=generate-travis
+install: pip install cram
+script: build/make-if-target-changed "$TARGET"
```

Comment:
I might inline a little bit of shell here instead. Would something like this be equivalent (enough) to what this separate script is doing?

```yaml
script: |
  git diff --name-only origin/master..HEAD | while read -r path; do
    target=$(basename "$path")/Dockerfile.tested
    [ -f "$target" ] && make "$target"
  done
```

Reply:
it's ~kinda different than that -- if you wanted it in one mega job you could kinda do it that way but then everything runs serially -- I wanted everything to run in parallel.

It's also not as simple as I'd like since the tests are in a different directory and you have to check both the tests and the actual contents (I did take a look at the attempt in circle, but it failed for the same reasons that this script does and the Makefile did)

Reply:
Oh right, I see. OK then, we'll roll with this for now.